app/services/geo_discovery/document_builder/wxs.rb

```diff
@@ -81,6 +81,15 @@ def generate_wms_path?
           @config && visibility && file_set && (raster_file_set? || vector_file_set?)
         end
 
+        # Determines if the wmts path should be generated
+        # @return [Boolean]
+        def generate_wmts_path?
+          return false unless visibility == "open"
+          return true if raster_set?
+          return true if raster_file_set?
+          resource_decorator.object.decorate.try(:mosaic_file_sets).present?
```

Comment:
Suggestion: Add `Wayfinder.for(resource).try(:mosaic_file_set_count)`, and then make that method use `custom_queries.find_deep_children_with_property(resource: resource, model: FileSet, property: :service_targets, value: "mosaic", count: true)` - that should be faster and more stable than serializing all the file sets into objects.

Reply:
Sounds good. Do we want to just change the one we added and continue going through the resource decorator? Or we aren't trying to contain use of wayfinders? There's a lot of using them via the resource decorators.

Reply:
Yeah that's a good call, let's continue using the decorator.